src/ledger/LedgerManagerImpl.cpp

```diff
@@ -530,34 +538,34 @@ LedgerManagerImpl::historyCaughtup(asio::error_code const& ec,
     }
     else
     {
-        // If we're in CATCHUP_RECENT mode, we actually only got part way
-        // through catchup -- we did the minimal prefix part -- and will
-        // get another callback as CATCHUP_COMPLETE when recent-replay is
-        // done. So for now just deposit LCL and prepare for replay.
-        if (mode == CatchupManager::CATCHUP_RECENT)
+        switch (progressState)
         {
-            mLastClosedLedger = lastClosed;
-            CLOG(INFO, "Ledger") << "First phase of CATCHUP_RECENT done: "
-                                 << ledgerAbbrev(mLastClosedLedger);
-            mCurrentLedger = make_shared<LedgerHeaderFrame>(lastClosed);
-            return;
-        }
+            case CatchupWork::ProgressState::APPLIED_BUCKETS:
+            {
+                mCurrentLedger = make_shared<LedgerHeaderFrame>(lastClosed.header);
```

Comment:
not necessarily for this PR (I see it was already like this), but it's a bit odd to have to update `mLastClosedLedger` here: I would have imagined that applying buckets would take care of it (just like replaying transactions indirectly updates it)